[Streams] Add data quality badge to Streams table

ElenaStoeva opened the pull request:
Closes https://github.com/elastic/kibana/issues/233966

## Summary

This PR adds a Data quality column to the Streams table.

<img width="1167" height="666" alt="Screenshot 2025-09-12 at 17 06 26" src="https://github.com/user-attachments/assets/5d1102ce-0b5d-41fa-ba6d-7ba8eae1ab1e" />

ElenaStoeva committed:
[Streams] Add data quality badge to Streams table

ElenaStoeva committed:
Merge branch 'main' into streams/data-quality-badge

ElenaStoeva committed:
Handle failed requests

elasticmachine commented:
Pinging @elastic/kibana-management (Team:Kibana Management)

ElenaStoeva committed:
Merge branch 'main' into streams/data-quality-badge

yngrdyn commented:
/oblt-deploy

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/339249)
* Commit: f88420e5390a2a198b4cd745f15dcb44b95f3184

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/339249#01994cca-355f-4eb7-baaa-889790b43dda)

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339249","number":339249,"commit":"f88420e5390a2a198b4cd745f15dcb44b95f3184"}],"number":339249}
buildkite-pr-comment-->

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

yngrdyn commented on x-pack/platform/plugins/shared/streams_app/public/components/stream_list_view/data_quality_badge.tsx:
Instead of creating this component, let's reuse the one in `dataset-quality` plugin: `QualityIndicator`

ElenaStoeva committed:
Reuse badge component from Data quality

ElenaStoeva committed:
Address feedback

ElenaStoeva commented on x-pack/platform/plugins/shared/streams_app/public/components/stream_list_view/tree_table.tsx:
```diff
@@ -145,6 +153,23 @@ export function StreamsTreeTable({
             ) : null,
         },
+        {
+          field: 'dataQiulity',
```
Addressed [5636dd5](https://github.com/elastic/kibana/pull/234946/commits/5636dd5ba004d2218a12683d895cb0fa4ee5d2f5)

ElenaStoeva committed:
Move indicator to package

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

ElenaStoeva committed:
Fix ci errors

ElenaStoeva committed:
Merge branch 'streams/data-quality-badge' of https://github.com/ElenaStoeva/kibana into streams/data-quality-badge

ElenaStoeva committed:
Revert "Move indicator to package"

This reverts commit 4d6ef15c147eef03ca3a47a0806c5b237bb732f4.

flash1293 commented:
> Should we show a count of 2 here? The change would be that for the histogram we query FROM <stream name>,<stream name>::failure.

@yngrdyn any opinion on this? We should probably keep in sync with the data set quality page

ElenaStoeva committed:
Address feedback

ElenaStoeva commented:
Thanks for the review @flash1293! I addressed your feedback around fetching the total doc count for both columns in the parent component.

> Not sure whether we can even fix this, but if there are no failures for a stream, the request throws an error:

This is because if there are no failed docs, there is no `::failures` index so the ESQL query fails as we try to fetch from a non-existing index. I don't think we can avoid this unless we use another query.

> Should we show a count of 2 here? The change would be that for the histogram we query FROM <stream name>,<stream name>::failure

The problem with this is that this query would fail if the there are no failed docs. We would need to fetch the failed docs in a separate query.

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/339809)
* Commit: 2de9eb87e8aa962cba55e382867368111b8d8bdc

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/339809#0199526e-730e-4468-b4ed-23c650c16324)

### History
* :broken_heart: [Build #339790](https://buildkite.com/elastic/kibana-pull-request/builds/339790) failed 14efc9b40ea9a9ff5ddf3c144f1808570b1b5b8a
* :broken_heart: [Build #339777](https://buildkite.com/elastic/kibana-pull-request/builds/339777) failed c8ab667d9d9b7699dccd644d03e2f7b00d1b3f62
* :broken_heart: [Build #339738](https://buildkite.com/elastic/kibana-pull-request/builds/339738) failed cd87ba5c2e92ded3e686be16897b24161074b68a
* :broken_heart: [Build #339462](https://buildkite.com/elastic/kibana-pull-request/builds/339462) failed e54eab44cd55351a17e3cc66c4c600c4210f5ef9
* :broken_heart: [Build #339362](https://buildkite.com/elastic/kibana-pull-request/builds/339362) failed 9a347541f50d29d7e103cb6ea2fe24fccef13a8e

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339809","number":339809,"commit":"2de9eb87e8aa962cba55e382867368111b8d8bdc"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339790","number":339790,"commit":"14efc9b40ea9a9ff5ddf3c144f1808570b1b5b8a"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339777","number":339777,"commit":"c8ab667d9d9b7699dccd644d03e2f7b00d1b3f62"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339738","number":339738,"commit":"cd87ba5c2e92ded3e686be16897b24161074b68a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339462","number":339462,"commit":"e54eab44cd55351a17e3cc66c4c600c4210f5ef9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339362","number":339362,"commit":"9a347541f50d29d7e103cb6ea2fe24fccef13a8e"}],"number":339809}
buildkite-pr-comment-->

ElenaStoeva committed:
Remove types from package

flash1293 commented:
> This is because if there are no failed docs, there is no ::failures index so the ESQL query fails as we try to fetch from a non-existing index. I don't think we can avoid this unless we use another query.

Let's keep it for now. I think we should fix this in Elasticsearch

> The problem with this is that this query would fail if the there are no failed docs. We would need to fetch the failed docs in a separate query.

I don't think it fails if you query both at the same time - if you do `FROM <name>::data,<name>::failures` in a single query, it won't fail even if there are no failed docs. If you only do `FROM <name>::failures`, only then the query fails.

ElenaStoeva committed:
Add failed docs to histogram query

ElenaStoeva committed:
Revert exporting to package

ElenaStoeva committed:
Merge branch 'main' into streams/data-quality-badge

ElenaStoeva committed:
Revert translation changes

ElenaStoeva committed:
Remove constants from package

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/339880)
* Commit: e81687a13b1f5fe7a6024373da427575a4a4ec37

### Failed CI Steps
* [Build Kibana Distribution](https://buildkite.com/elastic/kibana-pull-request/builds/339880#019952bd-166d-4c46-a21a-b9c6f6b4e1ad)

### History
* :broken_heart: [Build #339837](https://buildkite.com/elastic/kibana-pull-request/builds/339837) failed d480e5fb9207a1b993014d11e469e913ae94fa03
* :broken_heart: [Build #339820](https://buildkite.com/elastic/kibana-pull-request/builds/339820) failed 448f9d5d366de8df40d8cbfa702e73d24435619e
* :broken_heart: [Build #339809](https://buildkite.com/elastic/kibana-pull-request/builds/339809) failed 2de9eb87e8aa962cba55e382867368111b8d8bdc
* :broken_heart: [Build #339790](https://buildkite.com/elastic/kibana-pull-request/builds/339790) failed 14efc9b40ea9a9ff5ddf3c144f1808570b1b5b8a
* :broken_heart: [Build #339777](https://buildkite.com/elastic/kibana-pull-request/builds/339777) failed c8ab667d9d9b7699dccd644d03e2f7b00d1b3f62

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339880","number":339880,"commit":"e81687a13b1f5fe7a6024373da427575a4a4ec37"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339837","number":339837,"commit":"d480e5fb9207a1b993014d11e469e913ae94fa03"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339820","number":339820,"commit":"448f9d5d366de8df40d8cbfa702e73d24435619e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339809","number":339809,"commit":"2de9eb87e8aa962cba55e382867368111b8d8bdc"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339790","number":339790,"commit":"14efc9b40ea9a9ff5ddf3c144f1808570b1b5b8a"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339777","number":339777,"commit":"c8ab667d9d9b7699dccd644d03e2f7b00d1b3f62"}],"number":339880}
buildkite-pr-comment-->

ElenaStoeva committed:
Update bundle size limit

ElenaStoeva committed:
Merge branch 'main' into streams/data-quality-badge

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/339909)
* Commit: 878168b3f33461e007e73c9b63e3452287cd5357

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/eea4c86575ea10eb454a8513f0373ab38b65f81a) | [after](https://github.com/elastic/kibana/commit/878168b3f33461e007e73c9b63e3452287cd5357) | diff |
| --- | --- | --- | --- |
| `streamsApp` | 650 | 652 | +2 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/eea4c86575ea10eb454a8513f0373ab38b65f81a) | [after](https://github.com/elastic/kibana/commit/878168b3f33461e007e73c9b63e3452287cd5357) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 33 | 43 | +10 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/eea4c86575ea10eb454a8513f0373ab38b65f81a) | [after](https://github.com/elastic/kibana/commit/878168b3f33461e007e73c9b63e3452287cd5357) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 438.9KB | 437.5KB | -1.4KB |
| `streamsApp` | 677.4KB | 679.7KB | +2.3KB |
| total |  |  | +898.0B |

#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/eea4c86575ea10eb454a8513f0373ab38b65f81a) | [after](https://github.com/elastic/kibana/commit/878168b3f33461e007e73c9b63e3452287cd5357) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 11 | 12 | +1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/eea4c86575ea10eb454a8513f0373ab38b65f81a) | [after](https://github.com/elastic/kibana/commit/878168b3f33461e007e73c9b63e3452287cd5357) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 24.5KB | 26.0KB | +1.6KB |
| `streamsApp` | 12.1KB | 12.2KB | +143.0B |
| total |  |  | +1.7KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/eea4c86575ea10eb454a8513f0373ab38b65f81a) | [after](https://github.com/elastic/kibana/commit/878168b3f33461e007e73c9b63e3452287cd5357) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 33 | 43 | +10 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/eea4c86575ea10eb454a8513f0373ab38b65f81a) | [after](https://github.com/elastic/kibana/commit/878168b3f33461e007e73c9b63e3452287cd5357) | diff |
| --- | --- | --- | --- |
| `streamsApp` | 10 | 11 | +1 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/eea4c86575ea10eb454a8513f0373ab38b65f81a) | [after](https://github.com/elastic/kibana/commit/878168b3f33461e007e73c9b63e3452287cd5357) | diff |
| --- | --- | --- | --- |
| `streamsApp` | 22 | 23 | +1 |


</details>


### History
* :broken_heart: [Build #339880](https://buildkite.com/elastic/kibana-pull-request/builds/339880) failed e81687a13b1f5fe7a6024373da427575a4a4ec37
* :broken_heart: [Build #339837](https://buildkite.com/elastic/kibana-pull-request/builds/339837) failed d480e5fb9207a1b993014d11e469e913ae94fa03
* :broken_heart: [Build #339820](https://buildkite.com/elastic/kibana-pull-request/builds/339820) failed 448f9d5d366de8df40d8cbfa702e73d24435619e
* :broken_heart: [Build #339809](https://buildkite.com/elastic/kibana-pull-request/builds/339809) failed 2de9eb87e8aa962cba55e382867368111b8d8bdc
* :broken_heart: [Build #339790](https://buildkite.com/elastic/kibana-pull-request/builds/339790) failed 14efc9b40ea9a9ff5ddf3c144f1808570b1b5b8a

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339909","number":339909,"commit":"878168b3f33461e007e73c9b63e3452287cd5357"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339880","number":339880,"commit":"e81687a13b1f5fe7a6024373da427575a4a4ec37"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339837","number":339837,"commit":"d480e5fb9207a1b993014d11e469e913ae94fa03"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339820","number":339820,"commit":"448f9d5d366de8df40d8cbfa702e73d24435619e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339809","number":339809,"commit":"2de9eb87e8aa962cba55e382867368111b8d8bdc"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339790","number":339790,"commit":"14efc9b40ea9a9ff5ddf3c144f1808570b1b5b8a"}],"number":339909}
buildkite-pr-comment-->

flash1293 committed:
centralize fetching

flash1293 committed:
remove debug thingy

yngrdyn commented on x-pack/platform/plugins/shared/streams_app/public/hooks/use_streams_histogram_fetch.ts:
```diff
@@ -0,0 +1,115 @@
+        params: {
+          body: {
+            operationName: 'get_doc_count_for_stream',
+            query: `FROM ${streamName} | STATS doc_count = COUNT(*) BY @timestamp = BUCKET(@timestamp, ${minInterval} ms)`,
```
```suggestion
            query: `FROM ${streamName},${streamName}::failures | STATS doc_count = COUNT(*) BY @timestamp = BUCKET(@timestamp, ${minInterval} ms)`,
```

yngrdyn commented on x-pack/platform/plugins/shared/streams_app/public/components/stream_list_view/data_quality_column.tsx:
```diff
@@ -0,0 +1,77 @@
+    [allTimeseries]
+  );
+
+  const degradedDocCount = failedDocsResult?.value
```
```suggestion
  const degradedDocCount = degradedDocsResult?.value
```

yngrdyn commented on x-pack/platform/plugins/shared/streams/server/routes/internal/streams/crud/route.ts:
```diff
@@ -38,6 +39,10 @@ export const listStreamsRoute = createServerRoute({
 
     const streamNames = streams.filter(({ exists }) => exists).map(({ stream }) => stream.name);
 
+    const failureStorePrivileges = await scopedClusterClient.asCurrentUser.security.hasPrivileges({
```
I think this is not necessary, `ESQL` will return only results for the indices the user has privileges
For the case of failure stores, if they user doesn't have access it will return
<img width="2640" height="728" alt="image" src="https://github.com/user-attachments/assets/71462088-ae2a-40ed-963f-50b77ca64776" />
So potentially we could handle this same as the case where fs doesn't exists, in that case the number of documents will be `0`, and that won't add up to the totalDocs and hence it won't play a role in the percentage calc. That would make the code also cleaner because we won't need `considerFailedQuality` boolean flag in `DataQualityColumn` and also we could remove the branches there

ElenaStoeva committed:
Update x-pack/platform/plugins/shared/streams_app/public/components/stream_list_view/data_quality_column.tsx

Co-authored-by: Yngrid Coello <yngrdyn@gmail.com>

ElenaStoeva committed:
Update x-pack/platform/plugins/shared/streams_app/public/components/stream_list_view/data_quality_column.tsx

Co-authored-by: Yngrid Coello <yngrdyn@gmail.com>

ElenaStoeva committed:
Update x-pack/platform/plugins/shared/streams_app/public/hooks/use_streams_histogram_fetch.ts

Co-authored-by: Yngrid Coello <yngrdyn@gmail.com>

ElenaStoeva committed:
Update x-pack/platform/plugins/shared/streams_app/public/components/stream_list_view/data_quality_column.tsx

Co-authored-by: Yngrid Coello <yngrdyn@gmail.com>

ElenaStoeva committed:
Address feedback

ElenaStoeva commented:
Thanks for the review @yngrdyn! I addressed your comments and opened a follow-up issue https://github.com/elastic/kibana/issues/235337.

flash1293 committed:
fix type

ElenaStoeva committed:
Merge branch 'main' into streams/data-quality-badge

ElenaStoeva committed:
Increase bundle size limit for datasetQuality

yngrdyn reviewed (approved):
LGTM 🚀

ElenaStoeva committed:
Merge branch 'main' into streams/data-quality-badge

yngrdyn committed:
Merge branch 'main' into streams/data-quality-badge
